src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -258,14 +258,14 @@ public static Path of(URI uri) {
 
     /**
      * Returns the file name extension of this path as a {@code String}. The
-     * extension is defined to be the portion of the {@code String}
-     * representation of the file name after the last dot ('.'). All leading
-     * dots are ignored. If the extension is missing, then an
-     * {@link String#isEmpty() empty} {@code String} ({@code ""}) is returned.
-     * This will occur if the path has zero elements ({@link #getFileName()}
-     * returns {@code null}), the file name string does not contain a dot, or
-     * only the first character is a dot. If the last character is a dot, the
-     * extension is the empty string.
+     * extension is defined to be the portion of the
+     * {@link #getFileName file name} string after the last dot ('.'). All
+     * leading dots in the string are ignored. If the extension is missing,
```

Comment:
Simplify `All leading` to `Leading`.

Change `the extension is missing` (which suggests an error) to `there is no extension`.